lib/internal/validators.js

```diff
@@ -233,9 +229,17 @@ const validateAbortSignal = hideStackFrames((signal, name) => {
   }
 });
 
-const validateFunction = hideStackFrames((value, name) => {
-  if (typeof value !== 'function')
-    throw new ERR_INVALID_ARG_TYPE(name, 'Function', value);
+const validateFunction = hideStackFrames((value, name, allowClass = false) => {
+  if (typeof value === 'function') {
+    if (allowClass) return;
+
+    if (inspect === undefined)
+      ({ inspect } = require('util'));
+
+    if (!inspect(value).startsWith('[class')) return;
```

Comment:
I didn't notice this in my first review, I don't think we should do this, or at least not like this. This has probably performance implications, and doesn't seem very useful?

Reply:
This change was made to disallow classes from being passed in, and this should have a small impact on the performance since it won't inspect anything other than functions, which should be really fast, so I don't really see an issue here, I've also added a new parameter as suggested from one of the reviewers as to allow classes or not, since some validators also takes these kinds of parameters and options to allow certain things of the same type, such as the `validateObject()` validator as it gives an option to allow arrays as they're of the same type, so this should be alright.

Reply:
Please take a look at the benchmark results, it does have a huge perf impact on some critical chunks of the codebase.

Can you avoid force pushing so it's easier for reviewers to see what changed? This PR touches 79 files, having to go through each of them every time is quite cumbersome.

Reply:
> Please take a look at the benchmark results, it does have a huge perf impact on some critical areas of the codebase.

That's pretty strange I'd say, I've done some benchmarks outside of here to ensure there isn't much of a perf regression, as all it does is to just inspect a function/class, which should be fast, I don't see how it causes a perf regression here but I'm gonna leave that to you or the other reviewers to decide upon that.

Sure, sorry about that; I'll be pushing the new changes as individual commits to make reviewing easier since commits will be squashed at the end either way.

Reply:
I'm not really surprised, in my experience calling `inspect` is a relatively heavy task, and we're doing a substring comparison on top of that which is also not free.
I don't think it's a problem if the validator let `class` pass. I had a problem with `ERR_INVALID_CALLBACK` letting classes through because classes are not callbacks; a class is a function is JS, so I don't have an issue with `validateFunction` accepting classes.

I insist, I think `validateFunction` only check the `typeof`, any further checks is not worth it for the great majority of cases imo.

Reply:
Oh, when you said that it lets classes pass in the referenced PR, I thought you were also talking about the `validateFunction()` validator, I'll revert the changes to that validator and let it allow classes as a whole if that's the best approach here.